### ISSUE TITLE
Multiline support missing

### ISSUE THREAD

**Issue by Loremoth** [pygame/pygame#2574]
It seems very strange to me that a famous videogame library like this misses simple support for writing multiline text on screen. I hope it will be fixed soon...

**Comment by Starbuck5** [comment]
It's not like pygame has just forgotten to do this, it's more that "GUI" stuff has been seen as "out of scope" for pygame itself.

However, this could probably be doable. There could be a `wrap_pixel` keyword argument on font.render and freetypefont.render. But would it also have to handle `\n` and `\t` characters?

I wonder what people who have written GUI libraries for pygame think. Could this simplify / optimize your workflow? Should pygame add this? What sort of interface should it use? @ppizarror @MyreMylar

**Comment by ppizarror** [comment]
I think there's plenty of libraries that wrap text, for example ``textwrap`` (used by pygame-menu); having a simple core API is better because this kind of feature would not be used for many people, as in general a GUI library is preferred.

For example, if that ``wrap_pixel``is implemented, what is the distance between lines? (so, line height should also be provided). And about new lines (\n, \t), or language support (for word-breaking). And what about RTL-languages, and the text remaining these lines, they must be centered, justified, left, right? There's a lot of things to wonder. Generally this high-level features introduce lots of problems.

**Comment by BastiHz** [comment]
For anyone who wants to implement this for themselves there is a helpful example in the docs: https://www.pygame.org/docs/ref/freetype.html#pygame.freetype.Font.render_to

**Comment by robertpfeiffer** [comment]
In the past this would have been closed as wontfix, because it can easily be done in pure Python, and typesetting whole paragraphs is definitely out of scope.

**Comment by ankith26** [comment]
I agree, this kind of feature is probably best implemented in user code, or a 3rd party pygame library. pygame has always been a lower level tool which you can use to build your engine and game on, and this feature would need a higher level API, with the existing pygame API it would not be elegant to integrate this feature. There are many good libs that use pygame and provide this feature.

**Comment by Starbuck5** [comment]
I disagree, I think this functionality can and should be added into base pygame.

It's a big edge for users to go from rendering single lines of text to full GUI libraries. And it seems awfully inconvenient for users to have to reimplement the same functionality over and over, if they aren't aware of GUI libraries or don't wish to use them.

The fact that we have reference code on how to do this supports that it is a common operation that we can abstract.

I suggest a single additional optional argument to Font.render and freetype.Font.render (and variations, but hopefully they are centralized under the hood). It could be named something like `typesetting`, and would be a dictionary containing certain keys, settings.

**Comment by ankith26** [comment]
I see this is a pretty useful thing indeed, but I also believe that incase we add this to pygame, it would need a nicer, higher level API to make it more pleasant to work with, rather than adding more hacks to existing API. [pygame-text](https://github.com/cosmologicon/pygame-text) is a really nice module, solely dedicated to multiline font rendering. If pygame did add multiline support on its own, perhaps we could talk with the owners of this lib to have some integration with pygame, rather than having to re-invent the wheel again on our end. OFC, another solution is to just let things be as is, users who need this feature can easily use this library, or any other thing
